noodles-bed/src/ser.rs

```diff
@@ -60,551 +61,450 @@ impl<'a> ser::Serializer for &'a mut Serializer {
     type SerializeStruct = Self;
     type SerializeStructVariant = Self;
 
-    // Here we go with the simple methods. The following 12 methods receive one
-    // of the primitive types of the data model and map it to JSON by appending
-    // into the output string.
-    fn serialize_bool(self, v: bool) -> Result<()> {
-        self.output += if v { "true" } else { "false" };
-        Ok(())
+    fn serialize_bool(self, _v: bool) -> Result<()> {
+        unreachable!()
```

Comment:
I don't think this is really unreachable, e.g. a contrived example might be able to implement `BedN<3>` with and reach the `serialize_bool` during serialization. If you want to panic, I think a [`todo!`](https://doc.rust-lang.org/std/macro.todo.html) or [`unimplemented!`](https://doc.rust-lang.org/std/macro.unimplemented.html) would be better, as we may want to implement these functions later. Otherwise you could also return an `Err` such as `Err(ser::Error::custom("unsupported type"))`.

Same applies too all other the `unimplemented!` calls.